diff --git a/game-jam-94/Scenes/Mutations/Mutation.gd b/game-jam-94/Scenes/Mutations/Mutation.gd
new file mode 100644
index 0000000..1db026a
--- /dev/null
+++ b/game-jam-94/Scenes/Mutations/Mutation.gd
@@ -0,0 +1,26 @@
+# Class for mutations. All new mutations should extend this.
+# Provides virtual methods that mutations should optionally define behaviour for.
+# Extends Node2D as some mutations need to be aware of 2D space.
+
+class_name Mutation
+extends Node2D
+
+
+# Function called when a mutation is first applied to a blob.
+# Should apply any one-time changes.
+func _ready() -> void: pass
+
+
+# Function called each physics frame.
+# Should implement any physics behaviour.
+func _physics_process(delta: float) -> void: pass
+
+
+# Function called by brain when searching for things to do.
+# Should update state + memory accordingly.
+func brain_search(brain: Brain) -> void: pass
+
+
+# Function called by brain when deciding for things to do.
+# Should use brain state + memory and create tasks.
+func brain_think(brain: Brain) -> void: pass
diff --git a/game-jam-94/Scenes/Mutations/Mutation.gd.uid b/game-jam-94/Scenes/Mutations/Mutation.gd.uid
new file mode 100644
index 0000000..0c7bf68
--- /dev/null
+++ b/game-jam-94/Scenes/Mutations/Mutation.gd.uid
@@ -0,0 +1 @@
+uid://c3cjhj7k0xbia
diff --git a/game-jam-94/Scripts/Blob/blob_brain.gd b/game-jam-94/Scripts/Blob/blob_brain.gd
index 00c5d33..e2aee48 100644
--- a/game-jam-94/Scripts/Blob/blob_brain.gd
+++ b/game-jam-94/Scripts/Blob/blob_brain.gd
@@ -26,6 +26,8 @@ var boredom_factor: float = 10
 var current_vigilance: float = 0
 var vigilance_factor: float = 10
 
+var mutations: Array[Mutation] = []
+
 @export var BOREDOM_VARIANCE: float = 20
 @export var VIGILANCE_VARIANCE: float = 20
 
@@ -38,6 +40,8 @@ func _ready() -> void:
 	for key in STATS:
 		STATS[key] = randi_range(20, 100)
 		print(key, ": ", STATS[key])
+	
+	mutations.append(eyes)
 
 
 #func _unhandled_input(event: InputEvent) -> void:
@@ -61,60 +65,13 @@ func _physics_process(delta: float) -> void:
 		current_vigilance -= STATS["vigilance"]
 		vigilance_factor = randf_range(VIGILANCE_VARIANCE * 0.5, VIGILANCE_VARIANCE * 1.5)
 		on_search()
-		
-# This updates the visible objects dictionary
+
+
 func on_search() -> void:
-	var visible = []
-	if eyes != null:
-		visible = eyes.get_visible_things()
-	else:
-		for key in visible_objects:
-			visible_objects[key] = []
-		return
-	var food = []
-	var blobs = []
-	var obstacles = []
-	
-	for thing in visible:
-		if thing.is_in_group("food"):
-			food.append({
-				"object": thing,
-				"location": thing.position
-			})
-		if thing.is_in_group("blobs") and thing != BODY:
-			blobs.append({
-				"object": thing,
-				"location": thing.position
-			})
-			print(name, " saw a blob!")
-		if thing.is_in_group("nav_group"):
-			obstacles.append({
-				"object": thing,
-				"location": thing.position
-			})
-	
-	visible_objects = {
-		"food": food,
-		"blobs": blobs,
-		"obstacles": obstacles
-	}
-			
+	for mutation in mutations:
+		mutation.brain_search(self)
+
 
 func on_think() -> void:
-	# If we found food, pick a random food and path towards it. Otherwise, if we have found a blob path away from it. Otherwise path randomly.
-	if visible_objects["food"].size() > 0:
-		print(name, " is running towards food")
-		var chosen_food = visible_objects["food"].pick_random()
-		target_location = chosen_food["location"]
-		TASK_MANAGER.new_task(TASK_SEEK)
-	elif visible_objects["blobs"].size() > 0:
-		print(name, " is running from a blob")
-		var chosen_blob = visible_objects["blobs"].pick_random()
-		var chosen_dir = (BODY.position - chosen_blob["location"]).normalized()
-		target_location = BODY.position + (chosen_dir * randf_range(MIN_WANDER_RANGE, MAX_WANDER_RANGE))
-		TASK_MANAGER.new_task(TASK_SEEK)
-	else:
-		print(name, " is resorting to wandering in a random direction")
-		var chosen_dir = Vector2.from_angle(randf_range(-PI, PI))
-		target_location = BODY.position + (chosen_dir * randf_range(MIN_WANDER_RANGE, MAX_WANDER_RANGE))
-		TASK_MANAGER.new_task(TASK_SEEK)
+	for mutation in mutations:
+		mutation.brain_think(self)
diff --git a/game-jam-94/Scripts/Mutations/Eyes/eyes.gd b/game-jam-94/Scripts/Mutations/Eyes/eyes.gd
index dbaba7c..dcd56c1 100644
--- a/game-jam-94/Scripts/Mutations/Eyes/eyes.gd
+++ b/game-jam-94/Scripts/Mutations/Eyes/eyes.gd
@@ -1,21 +1,70 @@
-extends Node2D
+extends Mutation
 
 @export var SIGHT_RANGE: float = 400.0
 
 
-func _ready() -> void:
-	pass
+# Search for things we can see and add to brain's visible_objects
+func brain_search(brain: Brain) -> void:
+	var visible = _get_visible_things()
+
+	var food = []
+	var blobs = []
+	var obstacles = []
+	
+	for thing in visible:
+		if thing.is_in_group("food"):
+			food.append({
+				"object": thing,
+				"location": thing.position
+			})
+		elif thing.is_in_group("blobs") and thing != brain.BODY:
+			blobs.append({
+				"object": thing,
+				"location": thing.position
+			})
+			print(name, " saw a blob!")
+		elif thing.is_in_group("nav_group"):
+			obstacles.append({
+				"object": thing,
+				"location": thing.position
+			})
+	
+	brain.visible_objects = {
+		"food": food,
+		"blobs": blobs,
+		"obstacles": obstacles
+	}
+
+
+# If we found food, pick a random food and path towards it. Otherwise, if we have found a blob path away from it. Otherwise path randomly.
+func brain_think(brain: Brain) -> void:
+	if brain.visible_objects["food"].size() > 0:
+		print(name, " is running towards food")
+		var chosen_food = brain.visible_objects["food"].pick_random()
+		brain.target_location = chosen_food["location"]
+		brain.TASK_MANAGER.new_task(brain.TASK_SEEK)
+	elif brain.visible_objects["blobs"].size() > 0:
+		print(name, " is running from a blob")
+		var chosen_blob = brain.visible_objects["blobs"].pick_random()
+		var chosen_dir = (brain.BODY.position - chosen_blob["location"]).normalized()
+		brain.target_location = brain.BODY.position + (chosen_dir * randf_range(brain.MIN_WANDER_RANGE, brain.MAX_WANDER_RANGE))
+		brain.TASK_MANAGER.new_task(brain.TASK_SEEK)
+	else:
+		print(name, " is resorting to wandering in a random direction")
+		var chosen_dir = Vector2.from_angle(randf_range(-PI, PI))
+		brain.target_location = brain.BODY.position + (chosen_dir * randf_range(brain.MIN_WANDER_RANGE, brain.MAX_WANDER_RANGE))
+		brain.TASK_MANAGER.new_task(brain.TASK_SEEK)
 
 
 # Get all things our eyes can see
-func get_visible_things() -> Array:
+func _get_visible_things() -> Array:
 	var space := get_world_2d().direct_space_state
 	var hits := _find_objects_radius(space)
 	var visible := []
 	
 	for hit in hits:
 		var target = hit.collider
-		if _has_los(target.global_position, space):
+		if _has_los(target, space):
 			visible.append(target)
 	
 	return visible
@@ -30,14 +79,16 @@ func _find_objects_radius(space: PhysicsDirectSpaceState2D) -> Array[Dictionary]
 	vision_query.shape = vision_shape
 	vision_query.transform = global_transform
 	vision_query.collide_with_areas = true
+	vision_query.collide_with_bodies = true
+	vision_query.exclude = [get_parent().get_rid()]
 	
 	return space.intersect_shape(vision_query)
 
 
 # Check if eyes have clear line-of-sight to a target
-func _has_los(target: Vector2, space: PhysicsDirectSpaceState2D) -> bool:
-	var ray := PhysicsRayQueryParameters2D.create(global_position, target)
-	ray.exclude = [get_parent()]
+func _has_los(target: Node2D, space: PhysicsDirectSpaceState2D) -> bool:
+	var ray := PhysicsRayQueryParameters2D.create(global_position, target.global_position)
+	ray.exclude = [get_parent().get_rid(), target.get_rid()]
 	
 	var result := space.intersect_ray(ray)
 	return result.is_empty()
